### PR TITLE
A different xterminal-256 to RGB mapping

### PR DESCRIPTION
**Pull request by randrej**
The original color map reflects iTerm2's colors, which are not truly sRGB mapped like, say, urxvt's are. That's why GUI an terminal versions of [n]vim have different colors on Linux, which gets on my nerves. This patch adds a variable  `g:seoul256_true_srgb_mapping`, which changes the map to suit color rendering in Linux.

I added an explanation to the readme, too.

### CODE REVIEW

File: colors/seoul256.vim [hunk 1]
```diff
@@ -36,22 +36,106 @@
...
-  \ 143: '#BDBB72', 144: '#BDBC98', 145: '#BDBDBD', 151: '#BCDDBD', 152: '#BCDEDE',
-  \ 153: '#BCE0FF', 161: '#E12672', 168: '#E17899', 173: '#E19972', 174: '#E09B99',
-  \ 179: '#DFBC72', 181: '#E0BEBC', 184: '#DEDC00', 186: '#DEDD99', 187: '#DFDEBD',
-  \ 189: '#DFDFFF', 216: '#FFBD98', 217: '#FFBFBD', 218: '#FFC0DE', 220: '#FFDD00',
-  \ 222: '#FFDE99', 224: '#FFDFDF', 230: '#FFFFDF', 231: '#FFFFFF', 232: '#060606',
-  \ 233: '#171717', 234: '#252525', 235: '#333233', 236: '#3F3F3F', 237: '#4B4B4B',
-  \ 238: '#565656', 239: '#616161', 240: '#6B6B6B', 241: '#757575', 249: '#BFBFBF',
-  \ 250: '#C8C8C8', 251: '#D1D0D1', 252: '#D9D9D9', 253: '#E1E1E1', 254: '#E9E9E9',
-  \ 255: '#F1F1F1' }
+    if exists('g:seoul256_true_srgb_mapping') && g:seoul256_true_srgb_mapping
```
Comment (junegunn) [inline colors/seoul256.vim]: A more conventional way is to put this is `get(g:, 'seoul256_true_srgb_mapping', 0)`

File: colors/seoul256.vim [hunk 2]
```diff
@@ -308,7 +392,7 @@ call s:hi('GitGutterChange', [68, 68], [s:dark_bg + 1, s:light_bg - 2])
 call s:hi('GitGutterDelete', [161, 161], [s:dark_bg + 1, s:light_bg -2 ])
 call s:hi('GitGutterChangeDelete', [168, 168], [s:dark_bg + 1, s:light_bg - 2])
 
-" http://vim.wikia.com/wiki/Highlight_unwanted_spaces     
+" http://vim.wikia.com/wiki/Highlight_unwanted_spaces
```
Comment (junegunn) [inline colors/seoul256.vim]: Can you revert this hunk? It's unrelated to the patch and the trailing spaces are intentional.

**Comment by junegunn** [comment]
Thanks, I left a couple comments. Can you also show me a screenshot of the colors here?

**Comment by junegunn** [comment]
I also would like to suggest renaming the variable to `g:seoul256_srgb`.

File: colors/seoul256.vim (hunk 1 above)
Comment (randrej) [inline colors/seoul256.vim]: Sure. I'm rather new to vimscript.

File: colors/seoul256.vim (hunk 2 above)
Comment (randrej) [inline colors/seoul256.vim]: Absoulutely. I see the intent. I screwed up by adding the whole file to the commit (and had an autocommand that strips the whitespace on write).

**Comment by randrej** [comment]
OS: Arch Linux
WM: i3
image
images, left to right:
```
1st: g:seoul256_srgb=0, urxvt;
2nd: g:seoul256_srgb=0, nvim-qt;
3rd: g:seoul256_srgb=1, urxvt;
4th: g:seoul256_srgb=1, nvim-qt;
```

**Comment by randrej** [comment]
Fixed everything You said.

**Comment by junegunn** [comment]
Thanks but the trailing spaces are not restored.

**Comment by junegunn** [comment]
Merged, thanks!

**Pull request merged**